atsign-foundation/at_server
# fix: Do not terminate the server if there is an uncaught SocketException

**- What I did**
fix: Do not terminate the server if there is an uncaught SocketException; we only have uncaught SocketExceptions in weird edge cases

**- Additional context**
- Wrapping socket listening in runZonedGuarded blocks has not prevented this issue
- I continue to try to figure out how to resolve the underlying problem; meanwhile this PR will stop unnecessary atServer exits

**- How to verify it**
- Current automated tests pass
- Manual testing completed on a macbook where I can semi-reliably albeit painfully slowly reproduce the problem
  - Run a local atServer and some client program against it (e.g. an sshnpd)
  - Leave them running and put the macbook to sleep (e.g. close the lid)
  - Come back an hour later and open up the macbook; observe output of the server - usually the problem will have occurred
  - Observe the the client program continues to function correctly
  - Run other client programs e.g. repl doing lookups on other atServers
